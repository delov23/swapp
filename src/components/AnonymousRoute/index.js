export { default as AnonymousRoute } from './AnonymousRoute';
