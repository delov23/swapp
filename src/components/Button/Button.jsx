import React, { useContext } from 'react';

import styles from './Button.module.css';
import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';

const Button = props => {
  const theme = useContext(ThemeContext);

  return (
    <button className={styles[themify('submitBtn', theme)]} {...props}>
      {props.children}
    </button>
  );
};

export default Button;
