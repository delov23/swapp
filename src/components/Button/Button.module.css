.submitBtn {
  min-width: 20%;
  padding: 0.5rem;
  border-radius: 0.4rem;
  margin: 10px;
  color: var(--mainYellow);
  font-family: Arial;
  font-weight: 900;
  align-self: flex-end;
  font-size: 1rem;
  border: none;
  -webkit-border-radius: 0.4rem;
  -moz-border-radius: 0.4rem;
  -ms-border-radius: 0.4rem;
  -o-border-radius: 0.4rem;
}

.submitBtnLight {
  composes: submitBtn;
  background-color: var(--mainBlack);
}

.submitBtnDark {
  composes: submitBtn;
  background-color: var(--mainCyan);
}

@media screen and (max-width: 400px) {
  .submitBtn {
    min-width: 40%;
  }
}
