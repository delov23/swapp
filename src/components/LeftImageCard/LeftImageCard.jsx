import React, { useContext } from 'react';
import { Link } from 'react-router-dom';

import styles from './LeftImageCard.module.css';
import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';

const LeftImageCard = ({ width, image, text, href }) => {
  const theme = useContext(ThemeContext);

  return (
    <Link
      to={href}
      style={{ width }}
      className={styles[themify('card', theme)]}
    >
      <img className={styles.image} src={image} alt="" />
      <h3 className={styles.heading}>{text}</h3>
    </Link>
  );
};

export default LeftImageCard;
