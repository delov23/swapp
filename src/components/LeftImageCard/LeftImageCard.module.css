.card {
  width: 28vw;
  display: flex;
  justify-content: flex-start;
  margin: 5px;
  height: 5rem;
  align-items: center;
  border-width: 1px;
  border-style: solid;
  font-family: Starwars;
  border-radius: 10px;
  -webkit-border-radius: 10px;
  -moz-border-radius: 10px;
  -ms-border-radius: 10px;
  -o-border-radius: 10px;
}

.cardLight {
  composes: card;
  background-color: #fff;
  color: var(--mainCyan);
  border-color: #e5e9f2;
}

.cardDark {
  composes: card;
  background-color: #333;
  color: var(--mainYellow);
  border-color: #3c4858;
}

.image {
  height: 5rem;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.heading {
  font-family: Starwars;
  font-weight: 100;
  margin: auto;
  text-align: center;
}

@media screen and (max-width: 850px) {
  .card {
    width: 100% !important;
  }
}