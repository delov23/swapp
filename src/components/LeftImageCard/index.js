export { default as LeftImageCard } from './LeftImageCard';
