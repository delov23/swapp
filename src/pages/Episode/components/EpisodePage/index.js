export { default as EpisodePage } from './EpisodePage';
