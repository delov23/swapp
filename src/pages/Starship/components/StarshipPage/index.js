export { default as StarshipPage } from './StarshipPage';
