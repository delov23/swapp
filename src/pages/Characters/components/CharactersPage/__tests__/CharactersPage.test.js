import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import { CharactersPage } from '../';
import { LeftImageCard } from '../../../../../components/LeftImageCard';
import { Button } from '../../../../../components/Button';

const mockClick = jest.fn();

const EXAMPLE_PROPS = {
  pageInfo: {
    hasNextPage: true,
  },
  people: [
    {
      node: { name: 'Name 1', image: 'image 1', id: 'people.1' },
    },
    {
      node: { name: 'Name 2', image: 'image 2', id: 'people.2' },
    },
  ],
  handleClick: new mockClick(),
};

describe('<CharactersPage />', () => {
  it('should match snapshot', () => {
    expect(shallow(<CharactersPage {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should display info', () => {
    const wrapper = mount(
      <Router>
        <CharactersPage {...EXAMPLE_PROPS} />
      </Router>,
    );

    expect(wrapper.find(LeftImageCard)).toHaveLength(2);
  });

  it('should recognize click', () => {
    const wrapper = mount(
      <Router>
        <CharactersPage {...EXAMPLE_PROPS} />
      </Router>,
    );

    wrapper.find(Button).simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});
