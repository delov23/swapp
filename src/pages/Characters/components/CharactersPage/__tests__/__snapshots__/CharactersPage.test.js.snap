// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharactersPage /> should match snapshot 1`] = `
<main>
  <div
    className="cardsWrapper"
  >
    <div
      className="cards"
    >
      <LeftImageCard
        href="/characters/people.1"
        image="image 1"
        key="people.1"
        text="Name 1"
        width="20rem"
      />
      <LeftImageCard
        href="/characters/people.2"
        image="image 2"
        key="people.2"
        text="Name 2"
        width="20rem"
      />
    </div>
  </div>
  <Button>
    Load more
  </Button>
</main>
`;
